### PR TITLE
Sitemap dispatch event to add custom data // Rename dispatch event

### PR DESCRIPTION
**Pull request by bastienlm**
For understand this PR, see https://github.com/OpenMage/magento-lts/pull/248 :)
I update 'use exemple' : 

```
     /**
     * Add brand url to sitemap
     *
     * @event sitemap_urlset_generating_before
     * @param Varien_Event_Observer $observer
     */
    public function addBrandToSitemap(Varien_Event_Observer $observer)
    {
        $io               = $observer->getEvent()->getFile();
        $baseUrl    = $observer->getEvent()->getBaseUrl();
        $date         = $observer->getEvent()->getDate();
        $storeId    = $observer->getEvent()->getStoreId();

        $changefreq = (string)Mage::getStoreConfig('sitemap/brand/changefreq', $storeId);
        $priority   = (string)Mage::getStoreConfig('sitemap/brand/priority', $storeId);

        $collection = Mage::getResourceModel('brand/brand_collection')
            ->addActiveFilter()
        ;


        foreach ($collection as $brand) {
            $url = htmlspecialchars($baseUrl . Mage::helper('brand')->getBrandUrl($brand));

            if (Mage::getStoreConfig('web/seo/trailingslash'))
            {
                if (!preg_match('/\\.(rss|html|htm|xml|php?)$/', strtolower($url)) && substr($url, -1) != '/')
                {
                    $url .= '/';
                }
            }

            $xml = sprintf('<url><loc>%s</loc><lastmod>%s</lastmod><changefreq>%s</changefreq><priority>%.1f</priority></url>',
                $url,
                $date,
                $changefreq,
                $priority
            );
            $io->streamWrite($xml);
        }
        unset($collection);

    }

```

### CODE REVIEW

File: app/code/core/Mage/Sitemap/Model/Sitemap.php [hunk 1]
```diff
@@ -146,6 +146,13 @@ public function generateXml()
         $date    = Mage::getSingleton('core/date')->gmtDate('Y-m-d');
         $baseUrl = Mage::app()->getStore($storeId)->getBaseUrl(Mage_Core_Model_Store::URL_TYPE_LINK);
 
+        Mage::dispatchEvent('sitemap_urlset_generating_before', array(
```
Comment (LeeSaferite) [inline app/code/core/Mage/Sitemap/Model/Sitemap.php]: Why did you move the location of this event?

**Comment by LeeSaferite** [comment]
I commented on the code as well, but why did you move the location of the event dispatch? Based on what was commented on #248 this should have been a simple event rename. Instead it's an event rename *and* an event firing reorder.

File: app/code/core/Mage/Sitemap/Model/Sitemap.php (hunk 1 above)
Comment (colinmollenhour) [reply app/code/core/Mage/Sitemap/Model/Sitemap.php]: I had mentioned moving it in my original review so I suppose that is why. Reflecting back on that comment I don't think the order really matters

**Comment by bastienlm** [comment]
I've updated the PR and I just rename dispatch event (undo move)
